include/envoy/network/connection.h

```diff
@@ -233,6 +234,12 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * Get the socket options set on this connection.
    */
   virtual const ConnectionSocket::OptionsSharedPtr& socketOptions() const PURE;
+
+  /**
+   * @return Envoy::RequestInfo::FilterState& the state shared between filters operating
+   * on this connection.
+   */
+  virtual Envoy::RequestInfo::FilterState& perConnectionState() PURE;
```

Comment:
Design question: should this be a full RequestInfo, instead of just the FilterState?

Also, should this live on the ReadFilterCallbacks, instead of on the Connection?

Reply:
ReadFilterCallbacks has `connection()`, I think it is good to have this in Connection.

Reply:
RequestInfo does not seem to be related to FilterState, eventhough they are both under the same namespace. Doing so, would require exposing a RequestInfo accessor to get the Filterstate and making sure that the RequestInfo constructors will allocate the filterState as well.